File: triangle.php
<?php

/**
 * Author: Marcos Poletto Alves
 * Date: 2024-01-27
 */

function triangle($x) {

    // Define the first element. I'm supposing the triangle begin with it.
    $b[0][0] = 1;

    // loop through the cardinals until the number passed as argument
    for ($i=1; $i<=$x; $i++) {

        // Loop through the cardinals beginning with zero until argument.
        // Thus, we can use the left and right borders of the array to sum zero to them. 
        for ($j=0; $j<=$i; $j++) {

            // Test if values don't exist on the left and right could enable value changes to zero, 
            // keeping the entire logic of the triangle.
            $tmpa = !isset($b[$i-1][$j-1])? 0: $b[$i-1][$j-1];
            $tmpb = !isset($b[$i-1][$j])? 0: $b[$i-1][$j];
            $b[$i][$j] = (int)$tmpa + (int)$tmpb;

        }

    }

    return $b;

}

print_r(triangle(5));

File: triangle.py
#!/usr/bin/python3

'''
Author: Marcos Poletto Alves
Date: 2024-01-28
'''

def triangle(x):
    a = [[1]]
    for i in range(1, x+1):
        tmp = []
        for j in range(i):
            if j == 0:
                tl = 0 + a[i-1][j]
                tmp.append(tl)
            elif j < len(a[i-1]): 
                tm = a[i-1][j-1] + a[i-1][j]
                tmp.append(tm)
            
            if j == len(a[i-1])-1:
                tr = a[i-1][j] + 0
                tmp.append(tr)
        a.append(tmp)

    return a


print(triangle(10))